structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py

```
@@ -0,0 +1,818 @@
+"""Creep and shrinkage models MC2010 for concrete
+
+[1] fib Model Code for Concrete Structures 2010 (2013).
+
+TODO:
+ - Add the light-weight aggregate formulas
+ - Include the additional creep and shrinkage terms for other temperatures
+   (see 5.1.10.7 of [1])
+"""
+
+import numpy as np
+
+from structuralcodes.codes.mc2010._concrete_material_properties import (
+    _check_agg_types,
+    _get_Ecmod_coeffs,
+    _calc_E,
+    _calc_E28,
+    _check_cem_strength_class,
+)
+
+
+def _check_fcm(fcm: float) -> None:
+    """Check if the mean compressive strength is in the range of
+        applicability of the creep and shrinkage models:
+
+    Defined in fib Model Code 2010 (2013), section 5.1.9.4.2.
+
+    args:
+        fcm (float): The mean compressive strength of the concrete in
+            MPa.
+
+    returns:
+        Raises a ValueError if the mean compressive strength is outside
+            of the specified range.
+    """
+    if fcm < 20 or fcm > 130:
+        raise ValueError(
+            "The specified mean compressive strength is "
+            "outside of the range of applicability for the creep and "
+            "shrinkage laws given in the fib Model Code 2010. The mean"
+            " compressive strength has to be within the range of "
+            "20-130 MPa. Current compressive strength is"
+            f": {fcm} MPa."
+        )
+
+
+def _check_initial_stress(sigma: float, fcm: float) -> None:
+    """Check if the initial compressive stress (based on load at t0)
+        is within the range of applicability.
+
+    Defined in fib Model Code 2010 (2013), section 5.1.9.4.2.
+
+    args:
+        sigma (float): The compressive stress applied to the concrete
+            at t0 in MPa.
+        fcm (float): The mean compressive strength of the concrete in
+            MPa. Note that its value is non-negative.
+
+    returns:
+        Prints a warning if the initial compressive stress is
```

Raises a warning instead of print a warning